src/test/run-make/issue-11908/Makefile

```diff
@@ -12,12 +12,11 @@ all:
 	$(RUSTC) foo.rs --crate-type=dylib
 	mv $(call DYLIB,foo) $(TMPDIR)/other
 	$(RUSTC) foo.rs --crate-type=dylib
-	$(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | \
-		grep "multiple dylib candidates"
+	# dylibs fail with strange errors, but not assertions.
+	! $(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | grep "assertion failed"
```

Comment:
As per the comment -- this doesn't fail with "multiple dylib candidates" anymore (the dylibs have the same hash), but still fails for reasons I do not understand.